eclipse-scout-core/src/form/fields/smartfield/SmartField.ts

```diff
@@ -692,14 +692,35 @@ export class SmartField<TValue> extends ValueField<TValue> implements SmartField
       .then(({formattedValue, result}) => {
         this._triggerLookupCallDone(result);
         return formattedValue;
+      })
+      .catch(() => {
+        return this._invalidKeyLookup(value);
       });
   }
 
   protected _lookupByKeyDone(result: SmartFieldLookupResult<TValue>): string {
     this._notUnique = false;
     let lookupRow = LookupCall.firstLookupRow(result);
-    this.setLookupRow(lookupRow);
-    return this._formatLookupRow(lookupRow);
+    if (lookupRow) {
+      this.setLookupRow(lookupRow);
+      return this._formatLookupRow(lookupRow);
+    }
+
+    return this._invalidKeyLookup(result.key);
+  }
+
+  protected _invalidKeyLookup(key : TValue) : string {
+    // lookup call didn't return a result. Maybe the data belonging to the key has been deleted.
+    let invalidValueMessage = this.session.text('ui.InvalidValue');
+    this.setLookupRow(scout.create((LookupRow<TValue>), {
+      key: key,
+      text: invalidValueMessage
+    }));
+    this.setErrorStatus(scout.create(ValidationFailedStatus, {
```

Comment:
Actually, we should only use addErrorStatus to not remove the existing status, but since the smart field does not make use of that feature anyway, it is ok for now. We can refactor all setErrorStatus calls at once later